From 4a5d2fe375a6f125de80d0dd551a79d66c881dff Mon Sep 17 00:00:00 2001
From: Sanyog Kale <skale@habana.ai>
Date: Tue, 09 Jun 2020 08:30:24 +0300
Subject: [PATCH]  [SW-11875] goya2: ppboot: Change preboot load address

Currently, preboot load address is set to ICCM to run preboot.
Since ICCM needs to be freed up for running zephyr OS, this patch
add changes to change load address of preboot from ICCM to scratchpad SRAM
at offset of 0x1000.

Change-Id: I12dbd68589d390901b94415244224150c8fc80c3
---

diff --git a/goya2_arc/common.h b/goya2_arc/common.h
index 9894c42..1781fc0 100644
--- a/goya2_arc/common.h
+++ b/goya2_arc/common.h
@@ -16,7 +16,7 @@
 #define SP_SRAM_BASE		0x1BFE0000
 #define SP_SRAM_PREBOOT_OFFSET	0x1000
 #define PPBOOT_TEXT_BASE	(FW_SRAM_BASE + 0xC)
-#define PREBOOT_TEXT_BASE	0x8
+#define PREBOOT_TEXT_BASE	(SP_SRAM_BASE + SP_SRAM_PREBOOT_OFFSET + 0x8)
 
 #define FUSE_WORDS_PER_BANK	0x40
 #define I2C_SLAVE_ADDRESS	0x6A
diff --git a/goya2_arc_gnu/common.h b/goya2_arc_gnu/common.h
index d5968f9..bc04b5f 100644
--- a/goya2_arc_gnu/common.h
+++ b/goya2_arc_gnu/common.h
@@ -13,8 +13,10 @@
 
 #define PCIE_FW_SRAM_BASE	0x1BFF0000
 #define FW_SRAM_BASE		0x1B050000
+#define SP_SRAM_BASE		0x1BFE0000
+#define SP_SRAM_PREBOOT_OFFSET	0x1000
 #define PPBOOT_TEXT_BASE	(FW_SRAM_BASE + 0xC)
-#define PREBOOT_TEXT_BASE	0x8
+#define PREBOOT_TEXT_BASE	(SP_SRAM_BASE + SP_SRAM_PREBOOT_OFFSET + 0x8)
 
 #define FUSE_WORDS_PER_BANK	0x40
 #define I2C_SLAVE_ADDRESS	0x6A
